components/dashboard/src/usage/download/DownloadUsage.tsx

```diff
@@ -0,0 +1,135 @@
+
+    const handleDownload = useCallback(async () => {
+        if (!org) {
+            return;
+        }
+
+        toast(
+            <DownloadUsageToast
+                orgName={org?.slug ?? org?.id}
+                attributionId={attributionId}
+                startDate={startDate}
+                endDate={endDate}
+            />,
+            {
+                autoHide: false,
+            },
+        );
+    }, [attributionId, endDate, org, startDate, toast]);
+
+    return (
+        <Button type="secondary" onClick={handleDownload} className="flex flex-row" icon={<DownloadIcon />}>
+            Export as CSV
+        </Button>
+    );
+};
+
+type DownloadUsageToastProps = Props & {
+    orgName: string;
+};
+
+const DownloadUsageToast: FC<DownloadUsageToastProps> = ({ attributionId, endDate, startDate, orgName }) => {
+    const [progress, setProgress] = useState(0);
+
+    const queryArgs = useMemo(
+        () => ({
+            orgName,
+            attributionId: AttributionId.render(attributionId),
+            from: startDate.startOf("day").valueOf(),
+            to: endDate.endOf("day").valueOf(),
+            onProgress: setProgress,
+        }),
+        [attributionId, endDate, orgName, startDate],
+    );
+    const { data, error, isLoading, abort, remove } = useDownloadUsageCSV(queryArgs);
+
+    const saveFile = useCallback(() => {
+        if (!data || !data.blob) {
+            return;
+        }
+
+        saveAs(data.blob, data.filename);
+    }, [data]);
+
+    useEffect(() => {
+        return () => {
+            abort();
+            remove();
+        };
+        // eslint-disable-next-line react-hooks/exhaustive-deps
+    }, []);
+
+    if (isLoading) {
+        return (
+            <div className="space-y-2">
+                <div className="flex flex-row items-center space-x-2">
+                    <span>Preparing usage export</span>
+                </div>
+                <ProgressBar percent={progress} />
+            </div>
+        );
+    }
+
+    if (error) {
+        return (
+            <div>
+                <div className="flex flex-row items-start space-x-2">
+                    <ExclamationIcon className="mt-1 w-4 h-4" />
+                    <span>Error exporting your usage data:</span>
+                </div>
+                <pre className="mt-2 whitespace-normal text-sm">{error.message}</pre>
+            </div>
+        );
+    }
+
+    if (!data || !data.blob || data.count === 0) {
+        return <span>There are no usage records for that date range</span>;
+    }
+
+    const readableSize = prettyBytes(data.blob.size);
+    const formattedCount = Intl.NumberFormat().format(data.count);
+
+    return (
+        <div className="space-y-2">
+            <div className="flex flex-row items-center justify-between space-x-2">
+                <div className="flex flex-row items-center space-x-2">
+                    <img src={StatusDoneIcon} className="w-5 h-5" alt="Completed icon" />
```

Comment:
**thought**: I'd still think the icon is overloading the toast when used without aligning the text below, see https://github.com/gitpod-io/gitpod/pull/17791#discussion_r1212785134.